File: BubbleSort.java
public class BubbleSort {
	public static void main(String[] args){
		int[] myArray = new int[]{10,-3,7,2,5,9,4,6};
		System.out.println("Initial array.");
		for(int i = 0; i < myArray.length; i++){
			System.out.println(myArray[i]);
		}
		for (int lastUunsortedIndex = myArray.length -1; lastUunsortedIndex > 0; lastUunsortedIndex--){
			for (int i = 0; i < lastUunsortedIndex; i++){
				if (myArray[i] > myArray[i+1]){
					swap(myArray, i, i+1);
				}
			}
		}
		
		System.out.println("After sorting array.");
		for(int i = 0; i < myArray.length; i++){
			System.out.println(myArray[i]);
		}
	}
	
	public static void swap(int[] myArray, int i, int j){
		if(i == j){
			return;
		}
		
		int temp = myArray[i];
		myArray[i] = myArray[j];
		myArray[j] = temp;
	}
}

File: bubble_sort.go
package main

import "fmt"

func main(){
	dataInput := []int{4, 1, 5, 8, 9, 3, 2, 6, 7}
	bubbleSort(dataInput)
}

func bubbleSort(unsorted []int){
	fmt.Println("Bubble Sort")
	fmt.Printf("Unsorted items: %d\n", unsorted)
	fmt.Println()
	var i, j int
	for i = 0; i < (len(unsorted) - 1); i++ {
		for j = 1; j < (len(unsorted) - i); j++ {
			if unsorted[j-1] > unsorted[j] {
				temp := unsorted[j]
				unsorted[j] = unsorted[j-1]
				unsorted[j-1] = temp
			}
		}
		fmt.Printf("Unsorted array after %d pass is %d\n:", i+1, unsorted)
	}
}
